awx/ui_next/src/screens/Job/useWsJob.js

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import useWebsocket from '../../util/useWebsocket';
+import { JobsAPI } from '../../api';
+
+export default function useWsJob(initialJob) {
+  const { type } = useParams();
+  const [job, setJob] = useState(initialJob);
+  const lastMessage = useWebsocket({
+    jobs: ['status_changed'],
+    control: ['limit_reached_1'],
+  });
+
+  useEffect(() => {
+    setJob(initialJob);
+  }, [initialJob]);
+
+  useEffect(
+    function parseWsMessage() {
+      async function fetchJob() {
+        const { data } = await JobsAPI.readDetail(job.id, type);
+        setJob(data);
+      }
+
+      if (!job || lastMessage?.unified_job_id !== job.id) {
+        return;
+      }
+
+      if (
+        ['successful', 'failed', 'error', 'cancelled'].includes(
+          lastMessage.status
```

Comment:
> Once a job finishes running, the job is refreshed via the API so that we can display all data that comes available after a job finishes.

I like the approach of adding the auto-reload to the hook in order to fetch the summary data when a final status event is received.

However, the summary statistics may continue to update and may not be correct until the job details endpoint returns `"event_processing_finished": true,`. Depending on the scale of the job, this can happen _long_ after the final job status message is received. 

To detect when event processing is actually finished, we might need to enter a retry loop and poll the api with http requests until the `event_processing_finished` flag is `true` or some reasonable retry limit is reached.